### PR TITLE
Fixed different time stamp error and map view.

### PR DESCRIPTION
**Pull request by sreejakshetty**
#91 #78

Fixed all the indentation issues .
Tested on online server.
screenshot from 2016-06-15 07 47 07

### CODE REVIEW

**Comment by sreejakshetty** [comment]
#83 
#77 
Date/Time picker for edit shifts view is also added here .

**Comment by hongquan** [comment]
Still coding convention error

```
date("H:i",$end_time)
```

By the way, we will use MySQL's DATETIME type to store date & time data, not use integer, as mentioned here: https://github.com/fossasia/engelsystem/pull/96#issuecomment-226113867

Please update.

**Comment by sreejakshetty** [comment]
@hongquan Done!

File: includes/pages/admin_shifts.php [hunk 1]
```diff
@@ -161,7 +154,7 @@ function admin_shifts() {
       } else {
         $ok = false;
         error(_('Please select a mode for needed angels.'));
-      }
+        }
```
Comment (hongquan) [inline includes/pages/admin_shifts.php]: Why the brace is pushed in?

**Comment by hongquan** [comment]
@kamishettysreeja25 Have you tested after you change the data type to DATETIME?
I don't see any PHP code change after that Is it really working with the old PHP code?

**Comment by sreejakshetty** [comment]
@hongquan Yes, I have tested it after changing to datetime . I have intially changed to TIMESTAMP but DATETIME has more range of values .Both of them work the same for me .

File: includes/pages/user_shifts.php [hunk 2]
```diff
@@ -117,23 +119,49 @@ function user_shifts() {
...
+      if (isset($_REQUEST['start']) && $tmp = DateTime::createFromFormat("Y-m-d", trim($_REQUEST['start'])))
         $start = $tmp->getTimestamp();
       else {
         $ok = false;
-        $msg .= error(_("Please enter a valid starting time for the shifts."), true);
+        error(_('Please select a start date.'));
       }
-
-      if (isset($_REQUEST['end']) && $tmp = DateTime::createFromFormat("Y-m-d H:i", trim($_REQUEST['end'])))
+
```
Comment (hongquan) [inline includes/pages/user_shifts.php]: Still trailing space

Comment (sreejakshetty) [inline includes/pages/user_shifts.php]: Yes, I have followed the format . After every 

```
if{
//code
}else{
//code
}
```

they left a line for identation.
screenshot from 2016-06-15 16 49 35

Comment (hongquan) [inline includes/pages/user_shifts.php]: I don't complaint about new line. I complaint about "trailing space".
The blank line for separating block of codes should contain only "\n", not "\n space space space".

Comment (hongquan) [inline includes/pages/user_shifts.php]: selection_098

**Comment by hongquan** [comment]
Here is test result in my local:
1. I create a shift
   
   selection_096
2. I go to the list of shifts. No thing shows up:
   
   selection_097
3. I look into database, the two columns you added are not filled with data:

```
mysql quan@localhost:engelsystem> SELECT title, start, end, start_time, end_time FROM `Shifts`;
+---------+------------+------------+--------------+------------+
| title   |      start |        end |   start_time |   end_time |
|---------+------------+------------+--------------+------------|
| Ca sáng | 1466074800 | 1466082000 |       <null> |     <null> |
+---------+------------+------------+--------------+------------+
```

**Comment by hongquan** [comment]
What is the purpose of `start`, `end`, `start_time`, `end_time`?

**Comment by sreejakshetty** [comment]
screenshot from 2016-06-15 17 20 34
screenshot from 2016-06-15 17 20 20
-As the shift is created at 18:00 and the filter is from 18:26 . The shifts are not shown.We need to change the shifts time to able to view the shifts .

start -> start date 
start_time -> start time
end -> end date
end_time -> end time 
which are used in the date picker
screenshot from 2016-06-15 17 18 29

**Comment by hongquan** [comment]
The time "18:29" is set automatically when I visit the page. Why the filter is set with the value which doesn't produce result? If it is automatic, it should be  a value which can produce the result.

However, I change the filter and it still doesn't show shift

selection_099

The field `start_time` is of DATETIME type, it means the field already includes day and hour (date and time), then why is the `start` field still there?

**Comment by sreejakshetty** [comment]
screenshot from 2016-06-15 17 30 28
I think it is because of datetime datatype . There are even variables such as **start , end , created_at_timestamp , edited_at_timestamp** all these are **ints** as we only store timestamps which is an int and convert it into date format using **date()** function . This procedure is followed through out the code.Should I change the start_time and end_time data types to int as we store a integer ?

**Comment by hongquan** [comment]
I already give the link, where I explain the reason why we should use DATETIME type. Didn't you go to the link?

I can predict that after changing datatype in database schema, you will need to update PHP code. That is why I alerted:

> Have you tested after you change the data type to DATETIME?
> I don't see any PHP code change after that Is it really working with the old PHP code?

but instead of testing and updating as I remind, you just skip it and now you say "it is because of datetime datatype"?

**Comment by sreejakshetty** [comment]
According the code , the filter time is set to local time of germany and it starts from germany's current time.
screenshot from 2016-06-15 17 41 28

**Comment by hongquan** [comment]
> According the code , the filter time is set to local time of germany and it starts from germany's current time.

Let's create an issue to fix this. User doesn't know and doesn't care the time zone of the server. We cannot let user use a time to create a shift, then use a different time to find the shift.

**Comment by hongquan** [comment]
Btw, even after I set time to 00:00, I still don't see the shift in the calendar.

**Comment by sreejakshetty** [comment]
I have gone through the link . I will work on changing the datatype to datetime. I wanted to create a new issue for changing datatype to datetime, thats why I asked should I change the datatype to int .
After , I  changed the datatype to datetime . I was able to create the shifts but I didn't see the sql database . Now that I have seen sql database I taught there needs to be a change in php files. Sorry, I didn't notice it that time as I was able to view shifts which were previously created .
Now , I will change all the variable ( start , end, start_time, end_time ) datatypes to datetime  and make necessary changes to php files.

**Comment by sreejakshetty** [comment]
@hongquan Can I create a new issue for changing the datatype to datetime and work on that.Can we merge this PR by changing the datatype to int . I will create a new PR for fixing the issue of changing datatype of all the date variable used in the code.

**Comment by hongquan** [comment]
The new field is created in this PR, then create it with correct data type right now. Why do we have to create it with a certain type and then make another PR to convert it to correct type? If you can make something right, do it at once, don't make it incorrect then make 2nd step to correct it.

Btw, don't forget my comment:

> The field `start_time` is of DATETIME type, it means the field already includes day and hour (date and time), then why is the `start` field still there?

**Comment by sreejakshetty** [comment]
sure , I will make the changes in this PR only . 

> The field start_time is of DATETIME type, it means the field already includes day and hour (date and >time), then why is the start field still there?

I will consider the comment and see that do we require start_time variables as start variable can be used to store both the date and time .

**Comment by hongquan** [comment]
> I will consider the comment and see that do we require start_time variables as start variable can be used to store both the date and time

What I mean is to use 1 field, don't make duplication. For the name, should be easy to understand, less confusing.

**Comment by sreejakshetty** [comment]
Removed fields start_time and end_time from shifts table . Used start and end for both start time and end time to avoid duplication.
Tested on online server.

**Comment by hongquan** [comment]
Where is the change to MySQL's DATETIME data type?

**Comment by sreejakshetty** [comment]
I have not used start_time and end_time fields. I have removed those fields from shifts as they were not necessary and changed the files to original. So , I didn't change the datatype to DATETIME.

**Pull request merged**